src/main/java/ip/Duke.java

```diff
@@ -0,0 +1,339 @@
+                " What can I do for you?");
+        printHorizontalLine();
+    }
+
+    private static void printFarewell() {
+        printHorizontalLine();
+        System.out.println(" Bye. See you next time!");
+        printHorizontalLine();
+    }
+
+    private static void printInvalid() {
+        printHorizontalLine();
+        System.out.println(" ☹ OOPS!!! I'm sorry, but I don't know what that means :-(");
+        printHorizontalLine();
+    }
+
+    private static void printEmpty(String command) {
+        printHorizontalLine();
+        System.out.println(" ☹ OOPS!!! The description of a " + command + " cannot be empty.");
+        printHorizontalLine();
+    }
+
+    private static void printList(TaskManager taskManager) {
+        printHorizontalLine();
+        System.out.println("Here are the tasks in your list:");
+        taskManager.listTasks();
+        printHorizontalLine();
+    }
+
+    private static void printWrongFormatInteger() {
+        printHorizontalLine();
+        System.out.println(" The index of the task must be an integer.");
+        printHorizontalLine();
+    }
+
+    private static void printWrongFormatTask(String command, String parameter) {
+        printHorizontalLine();
+        System.out.println(" " + command + " requires the " + parameter + " parameter");
+        printHorizontalLine();
+    }
+
+    private static void printInvalidTask() {
+        printHorizontalLine();
+        System.out.println(" The task ID does not exist!");
+        printHorizontalLine();
+    }
+
+    private static void printFullTasks() {
+        printHorizontalLine();
+        System.out.println(" Sorry, unable to add any more tasks! The task manager is full. ");
+        printHorizontalLine();
+    }
+
+
+    /**
+     * Executes appropriate methods based on the given command.
+     *
+     * @param taskManager TaskManager object.
+     * @param line Line of user input.
+     */
+    private static void execute(TaskManager taskManager, String line) {
+        // Prevent blank tasks
+        if (line.isBlank()) {
+            return;
+        }
+
+        // Find position of first space
+        int spacePos = line.indexOf(" ");
+        // command is first word
+        String command = spacePos > 0 ? line.substring(0, spacePos).strip() : line;
+        String num;
+
+        switch (command) {
+        case COMMAND_LIST:
+            printList(taskManager);
+            break;
+        case COMMAND_DONE:
+            num = line.substring(spacePos + 1).strip();
+            execDone(taskManager, num);
+            break;
+        case COMMAND_BYE:
+            break;
+        case COMMAND_TODO: // Fallthrough
+        case COMMAND_DEADLINE: // Fallthrough
+        case COMMAND_EVENT:
+            try {
+                execAddTask(taskManager, command, line);
+            } catch (ArrayIndexOutOfBoundsException e) {
+                printFullTasks();
+            }
+            break;
```

Comment:
Nice use of constants in the switch block!

Reply:
I sort of learned from the [`Contacts1` `class`](https://github.com/nus-cs2113-AY2021S1/contacts/blob/master/src/main/java/Contacts1.java) in lecture. 😄